app/models/puppetclass.rb

```diff
@@ -202,7 +202,7 @@ def self.prepare_rdoc(root)
 
   def self.search_by_host(key, operator, value)
     conditions = sanitize_sql_for_conditions(["hosts.name #{operator} ?", value_to_sql(operator, value)])
-    direct     = Puppetclass.joins(:hosts).where(conditions).uniq.pluck('puppetclasses.id')
+    direct     = Puppetclass.joins(:hosts).where(conditions).pluck('puppetclasses.id').uniq
```

Comment:
Apparently calling `.pluck` after `.uniq` on a join table generates a query on Postgres that raises this exception `for SELECT DISTINCT, ORDER BY expressions must appear in select list`.